src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -978,6 +980,50 @@ public void UpdatePatchDetails(Patch patch)
         UpdateReportTabPatchName(TranslationServer.Translate(patch.Name));
     }
 
+    /// <summary>
+    ///   Setup/updates the existing part selection buttons with the correct properties based on their Node names.
+    /// </summary>
+    public void UpdateMicrobePartSelections()
+    {
+        foreach (var entry in placeablePartSelectionElements)
+        {
+            if (!SimulationParameters.Instance.DoesOrganelleExist(entry.Name))
+                continue;
+
+            var organelle = SimulationParameters.Instance.GetOrganelleType(entry.Name);
+
+            entry.PartName = organelle.Name;
+            entry.MPCost = organelle.MPCost;
+            entry.PartIcon = organelle.LoadedIcon;
+
+            if (!entry.IsConnected(
+                nameof(MicrobePartSelection.OnPartSelected), this, nameof(OnOrganelleToPlaceSelected)))
+            {
+                entry.Connect(
+                    nameof(MicrobePartSelection.OnPartSelected), this, nameof(OnOrganelleToPlaceSelected));
+            }
```

Comment:
I mean wouldn't it be much cleaner to have a separate method called by the editor ready method to do this setup just once. In my opinion it's much cleaner to separate initialization from updateable properties. And you could anyway have _Ready call:
```
SetupMicrobePartSelections();
UpdateMicrobePartSelections();
```